pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -149,12 +243,13 @@ func (c *ReplicaCalculator) GetResourceReplicas(currentReplicas int32, targetUti
 	return int32(math.Ceil(newUsageRatio * float64(len(metrics)))), utilization, timestamp, nil
 }
 
-// GetMetricReplicas calculates the desired replica count based on a target resource utilization percentage
-// of the given resource for pods matching the given selector in the given namespace, and the current replica count
-func (c *ReplicaCalculator) GetMetricReplicas(currentReplicas int32, targetUtilization float64, metricName string, namespace string, selector labels.Selector) (replicaCount int32, utilization float64, timestamp time.Time, err error) {
+// GetMetricReplicas calculates the desired replica count based on a target metric utilization
+// (as a milli-value) for pods matching the given selector in the given namespace, and the
+// current replica count
+func (c *ReplicaCalculator) GetMetricReplicas(currentReplicas int32, targetUtilization int64, metricName string, namespace string, selector labels.Selector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
```

Comment:
It seems that methods GetMetricReplicas and GetRawResourceReplicas are similar. Did you consider making their implementation to use a common method?